allennlp/common/params.py

```diff
@@ -296,15 +294,8 @@ def get(self, key: str, default: Any = DEFAULT):
         Performs the functionality associated with dict.get(key) but also checks for returned
         dicts and returns a Params object in their place with an updated history.
         """
-        if default is self.DEFAULT:
-            try:
-                value = self.params.get(key)
-            except KeyError:
-                raise ConfigurationError(
-                    'key "{}" is required at location "{}"'.format(key, self.history)
-                )
```

Comment:
This will never get called. `get` doesn't throw a `KeyError`. Am I missing something? It does apply to `pop` tho.

Reply:
No, I think you're right to remove this.  It was probably there because of a copy-paste a long time ago.